agent-inject/agent/config.go

```diff
@@ -215,9 +215,42 @@ func (a *Agent) newConfig(init bool) ([]byte, error) {
 		}
 	}
 
+	// If EnableQuit has been set, set it on the listener. If a listener hasn't
+	// been defined, set it on a new one. Also add a simple cache stanza since
+	// that's required for an agent listener.
+	if a.EnableQuit != nil {
+		if len(config.Listener) > 0 {
+			config.Listener[0].AgentAPI = &AgentAPI{
```

Comment:
In the vast majority of cases, there will only be 1 listener, but should we maybe set it on all listeners? Just choosing the first one seems surprising, and for users who want to finesse which listeners it is and isn't on, I think it's ok to not give them a dedicated annotation for that but let them fully define their own config.

Reply:
I don't think the injector can currently create a config with more than one listener, can it? I think the only way to inject an agent that has multiple listener is to specify the full config in a configmap, and in that case [this code wouldn't be triggered](https://github.com/hashicorp/vault-k8s/blob/afa2571312f017efb944a504abc9603a1f1343b3/agent-inject/agent/container_env.go#L49-L61).

But I can add a loop since it does look a little jarring to only reference listener[0] here.

Reply:
Oh I forgot about that - makes sense.